src/macros.js

```diff
@@ -33,86 +29,99 @@ const customFields = function(mediainfo, macros, customFieldsName) {
+      macros[`{mediainfo.${prop}}`] = defaults[`{mediainfo.${prop}}`];
+    } else {
+      macros[`{mediainfo.${prop}}`] = '';
+    }
+  });
 
-  // Get macros with defaults e.g. {x=y}, store values and replace with standard macros
-  string = string.replace(/{([^}=]+)=([^}]*)}/g, function(match, name, defaultVal) {
-    defaults[`{${name}}`] = defaultVal;
+  ['custom_fields', 'customFields'].forEach((customFieldProp) => {
+    customFields(mediainfo, macros, customFieldProp);
+  });
+
+  return macros;
+};
 
+const getDefaultValues = function(string) {
+  const defaults = {};
+  const modifiedString = string.replace(/{([^}=]+)=([^}]*)}/g, (match, name, defaultVal) => {
+    defaults[`{${name}}`] = defaultVal;
     return `{${name}}`;
   });
 
-  if (uriEncode === undefined) {
-    uriEncode = false;
-  }
+  return {defaults, modifiedString};
+};
 
-  let macros = {};
+const processMacroNameOverrides = function(string, macroNameOverrides) {
+  const foundOverrides = {};
 
-  if (customMacros !== undefined) {
-    macros = customMacros;
+  for (const defaultMacro in macroNameOverrides) {
+    const customMacro = macroNameOverrides[defaultMacro];
+
+    if (string.includes(customMacro)) {
+      foundOverrides[defaultMacro] = customMacro;
+      string = string.replace(new RegExp(customMacro, 'g'), defaultMacro);
+    }
   }
 
-  // Static macros
-  macros['{player.id}'] = this.options_['data-player'] || this.id_;
-  macros['{player.height}'] = this.currentHeight();
-  macros['{player.width}'] = this.currentWidth();
-  macros['{player.heightInt}'] = Math.round(this.currentHeight());
-  macros['{player.widthInt}'] = Math.round(this.currentWidth());
-  macros['{mediainfo.id}'] = this.mediainfo ? this.mediainfo.id : '';
-  macros['{mediainfo.name}'] = this.mediainfo ? this.mediainfo.name : '';
-  macros['{mediainfo.duration}'] = this.mediainfo ? this.mediainfo.duration : '';
-  macros['{player.duration}'] = this.duration();
-  macros['{player.durationInt}'] = Math.round(this.duration());
-  macros['{player.pageUrl}'] = videojs.dom.isInFrame() ? document.referrer : window.location.href;
-  macros['{playlistinfo.id}'] = this.playlistinfo ? this.playlistinfo.id : '';
-  macros['{playlistinfo.name}'] = this.playlistinfo ? this.playlistinfo.name : '';
-  macros['{timestamp}'] = new Date().getTime();
-  macros['{document.referrer}'] = document.referrer;
-  macros['{window.location.href}'] = window.location.href;
-  macros['{random}'] = Math.floor(Math.random() * 1000000000000);
+  return {string, foundOverrides};
+};
 
-  ['description', 'tags', 'reference_id', 'ad_keys'].forEach((prop) => {
-    if (this.mediainfo && this.mediainfo[prop]) {
-      macros[`{mediainfo.${prop}}`] = this.mediainfo[prop];
-    } else if (defaults[`{mediainfo.${prop}}`]) {
-      macros[`{mediainfo.${prop}}`] = defaults[`{mediainfo.${prop}}`];
-    } else {
-      macros[`{mediainfo.${prop}}`] = '';
-    }
-  });
+const getStaticMacros = function() {
```

Comment:
suggestion/non-blocking: I would suggest this function take a `player` argument instead of expecting it to be invoked via `call()`, but that's very minor for an internal function.